File: CommonChild.py
# https://www.hackerrank.com/challenges/common-child/problem?isFullScreen=true


def commonChild(s1 , s2):
    # Write your code here   
    last_row = [0]*(len(s1)+1)

    for i in range(1, len(s1)+1):
        current = [0]
        for j in range(1, len(s2)+1):
            if s1[i-1] == s2[j-1]:
                current.append(last_row[j-1]+1)
            else:
                current.append(max(last_row[j], current[-1]))
        last_row = current

    return last_row[-1]

File: CommonChild.cs
# https://www.hackerrank.com/challenges/common-child/problem?isFullScreen=true


 public static int commonChild(string s1, string s2)
    {
        char[] first = s1.ToCharArray();
        char[] second = s2.ToCharArray();
        int n = second.Length;
        
        int[] lcs = new int[second.Length+1];
        
        for(int i=1; i<=first.Length; i++){
            int current = 0;
            for(int j=1; j<=second.Length; j++){
                int temp = lcs[j];
                if(first[i-1] == second[j-1]){
                    lcs[j] = current + 1;
                }else {
                    lcs[j] = Math.Max(lcs[j], lcs[j-1]);
                }
                current = temp;
            }
        }
        
        return lcs[n];
    }

